zrml/prediction-markets/src/lib.rs

```diff
@@ -1191,19 +1220,57 @@ mod pallet {
                 Self::process_subsidy_collecting_markets(now, T::MarketCommons::now());
 
             let _ = with_transaction(|| {
-                let close = Self::market_close_manager(now, |market_id, market| {
-                    let weight = Self::on_market_close(market_id, market)?;
-                    total_weight = total_weight.saturating_add(weight);
-                    Ok(())
-                });
+                // If we are at genesis or the first block the timestamp is be undefined. No
+                // market needs to be opened or closed on blocks #0 or #1, so we skip the
+                // evaluation. Without this check, new chains starting from genesis will hang up,
+                // since the loops in `market_open_manager` and `market_close_manager` below will
+                // run over an interval of 0 to the current time frame.
+                if now <= 1u32.into() {
+                    return TransactionOutcome::Commit(Ok(()));
+                }
```

Comment:
Not really a noteworthy difference but we can move this out of the transactional body.

Reply:
:thumbsup:

Reply:
Mitigated in e1168926476897ba86cc50dadd536a538b10d3d9